iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachLoad.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Report;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.NetworkModification;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.BusbarSectionPosition;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.iidm.network.extensions.ConnectablePositionAdder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import static com.powsybl.iidm.modification.ModificationUtils.throwExceptionOrLogError;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.*;
+
+/**
+ * This method adds a new load on an existing voltage level. The voltage level should be described
+ * in node/breaker topology.
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+public class AttachLoad implements NetworkModification {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttachLoad.class);
+
+    private final LoadAdder loadAdder;
```

Comment:
Should be a Function<VoltageLevel, LoadAdder> 
otherwise we could write dump and not working code:
```java
n1 =  ...
n2 = ...
adder = n1.getVoltageLevel('vl').newLoadAdder()
   .setId("x")
   ...
   .add();
new AttachLoad(adder, ...)
  .apply(n2);
```

Reply:
Couldn't we check that the network of the `Load` is the same as the one passed in `apply`? Unfortunately this requires to create the load, as we don't have access to the adder parameters. We need to find a way to avoid the function, as if we think about serializing the `AttachLoad`, how do we deal with the `Function<VoltageLevel, LoadAdder>`?

Reply:
Why not completely add the adder from the user, and expose in this class only the methods that are still relevant?
I didn't look at the code in details, but for ex. I guess that methods such as setNode() or add() should not be called in those use cases:
hiding the adder would simply make it impossible for the user to do it, which would be more clear.

Reply:
The idea was to not duplicate the adder code in network modification.
Load creation is not a good example as it is very simple, but for a transformer for instance where you have a complex logic with tap changer, limits it would be a pity a replicate all of this here.
Also basing everything on adder would allow to have only one network modification for all injections, one for all branches. So InjectionFeederCreation (is should be renamed with something like this), would be a very short and simple code that would allow to create and connect all possible injection.

setNode could be call outside, BUT it will be rewritten inside 'modify' method before calling 'add' so this is not really a problem.

I am not sure to understand what you mean by hiding the adder.

The only drawback relying on adder provider is that is does not allow to serialize the whole modification (which is probably unacceptable). So we have to think about a solution for that.

Reply:
> Couldn't we check that the network of the `Load` is the same as the one passed in `apply`

That would be a design flaw

Reply:
> Couldn't we check that the network of the `Load` is the same as the one passed in `apply`? Unfortunately this requires to create the load, as we don't have access to the adder parameters. We need to find a way to avoid the function, as if we think about serializing the `AttachLoad`, how do we deal with the `Function<VoltageLevel, LoadAdder>`?

Yes this is the worried part.

Reply:
_Still about using the adder:_
actually it seems that at some point we will need to have `NetworkModification`s for adding elements anyway, in which we will need to store all the data of the element ?
So I think that at some point we will anyway need some POJO representation of all those attributes, separate from the adders, no ?

(by hinding the adder, I meant not exposing it in the API of the class, but maybe use it internally, for ex. having a `setId` which would call the `setId` of the adder, but finally I am not sure it's a good idea either, because of the serialization issue)

_About the re-use of one `InjectionFeederCreation`_, I understand the idea, but I am not sure it's very interesting since you need to specify their "connection point" individually anyway.

Reply:
I'm afraid that for making it serializable we will need to somehow duplicate adders with simple pojos not dependent of iidm objects. Let's see that in next pr as it is an big work for having it for all equipments.